fastlane/lanes/localization.rb

```diff
@@ -379,4 +381,20 @@
       lint_task: 'lintWordpressVanillaRelease' # TODO: Should we adapt this?
     )
   end
+
+  # Updates the `.po` file at the given `po_path` using the content of the `sources` files, interpolating `release_version` where appropriate.
+  # Internally, this calls the `an_update_metadata_source` release toolkit action and adds Git management to it.
+  #
+  def update_po_file_for_metadata_localization(po_path:, sources:, release_version:, commit_message:)
+    ensure_git_status_clean
+
+    an_update_metadata_source(
+      po_file_path: po_path,
+      source_files: sources,
+      release_version: release_version
+    )
+
+    git_add(path: po_path)
+    git_commit(path: po_path, message: commit_message, allow_nothing_to_commit: true)
```

Comment:
I debated whether to have `allow_nothing_to_commit: false` instead, to signal the caller that nothing happened, which should be unusual for a call to updated the `.po` file.

Ultimately, I decided against it to keep the flow unsurprising. Presumably, one would notice that nothing happens once they see no commits about the `.po` file in their history.

But, I'm on the edge. I could be easily persuaded that having the commit fail could be a useful signal for the developer calling the action.

Reply:
>  I could be easily persuaded that having the commit fail could be a useful signal for the developer calling the action.

I tried that out for size just now. One thing I don't like is how `git_commit` fails: With a stacktrace instead neat error capturing:

```
[22:15:15]: ------------------------
[22:15:15]: --- Step: git_commit ---
[22:15:15]: ------------------------
[22:15:15]: $ git commit -m Update\ WordPress\ \`PlayStoreStrings.po\`\ for\ version\ 19.8 /Users/gio/Developer/a8c/wpandroid/fastlane/../WordPress/metadata/PlayStoreStrings.po
[22:15:15]: ▸ On branch test/gio-16404
[22:15:15]: ▸ Your branch and 'origin/test/gio-16404' have diverged,
[22:15:15]: ▸ and have 3 and 2 different commits each, respectively.
[22:15:15]: ▸ (use "git pull" to merge the remote branch into yours)
[22:15:16]: ▸ nothing to commit, working tree clean
+-----------------------------+---------------------------------+
|                         Lane Context                          |
+-----------------------------+---------------------------------+
| DEFAULT_PLATFORM            | android                         |
| PLATFORM_NAME               | android                         |
| LANE_NAME                   | android update_appstore_strings |
| GIT_REPO_WAS_CLEAN_ON_START | true                            |
+-----------------------------+---------------------------------+
[22:15:16]: Exit status of command 'git commit -m Update\ WordPress\ \`PlayStoreStrings.po\`\ for\ version\ 19.8 /Users/gio/Developer/a8c/wpandroid/fastlane/../WordPress/metadata/PlayStoreStrings.po' was 1 instead of 0.
On branch test/gio-16404
Your branch and 'origin/test/gio-16404' have diverged,
and have 3 and 2 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean


+------+------------------------------------+-------------+
|                    fastlane summary                     |
+------+------------------------------------+-------------+
| Step | Action                             | Time (in s) |
+------+------------------------------------+-------------+
| 1    | default_platform                   | 0           |
| 2    | is_ci                              | 0           |
| 3    | is_ci                              | 0           |
| 4    | check_for_toolkit_updates          | 2           |
| 5    | Switch to android                  | 0           |
|      | update_wordpress_appstore_strings  |             |
|      | lane                               |             |
| 6    | ensure_git_status_clean            | 0           |
| 7    | an_update_metadata_source          | 0           |
| 8    | git_add                            | 0           |
| 💥   | git_commit                         | 0           |
+------+------------------------------------+-------------+

[22:15:16]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

bundler: failed to load command: fastlane (/Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/bin/fastlane)
Traceback (most recent call last):
	51: from /Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `<main>'
	50: from /Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `load'
	49: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/exe/bundle:36:in `<top (required)>'
	48: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	47: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/exe/bundle:48:in `block in <top (required)>'
	46: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli.rb:25:in `start'
	45: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	44: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli.rb:31:in `dispatch'
	43: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	42: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	41: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	40: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli.rb:484:in `exec'
	39: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli/exec.rb:23:in `run'
	38: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli/exec.rb:58:in `kernel_load'
	37: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli/exec.rb:58:in `load'
	36: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/bin/fastlane:23:in `<top (required)>'
	35: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/bin/fastlane:23:in `load'
	34: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/bin/fastlane:23:in `<top (required)>'
	33: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	32: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	31: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
	30: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	29: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	28: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	27: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	26: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	25: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	24: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	23: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	22: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	21: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	20: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	19: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	18: from lanes/localization.rb:91:in `block (2 levels) in parsing_binding'
	17: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	16: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'
	15: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'
	14: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	13: from lanes/localization.rb:123:in `block (2 levels) in parsing_binding'
	12: from lanes/localization.rb:398:in `update_po_file_for_metadata_localization'
	11: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	10: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	 9: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	 8: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	 7: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	 6: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	 5: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	 4: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/actions/git_commit.rb:18:in `run'
	 3: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/helper/sh_helper.rb:12:in `sh'
	 2: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane/lib/fastlane/helper/sh_helper.rb:80:in `sh_control_output'
	 1: from /Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/Users/gio/Developer/a8c/wpandroid/vendor/bundle/ruby/2.7.0/gems/fastlane-2.205.0/fastlane_core/lib/fastlane_core/ui/interface.rb:153:in `shell_error!': \e[31m[!] Exit status of command 'git commit -m Update\\ WordPress\\ \\`PlayStoreStrings.po\\`\\ for\\ version\\ 19.8 /Users/gio/Developer/a8c/wpandroid/fastlane/../WordPress/metadata/PlayStoreStrings.po' was 1 instead of 0. (FastlaneCore::Interface::FastlaneShellError)
On branch test/gio-16404
Your branch and 'origin/test/gio-16404' have diverged,
and have 3 and 2 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean
\e[0m
```

Reply:
This is strange, I remember a while ago testing `allow_nothing_to_commit: true` when working on another action and I don't remember it printing the stacktrace when there was indeed nothing to commit.

I wonder if the crash you had could be related to something else (even maybe the fact that your branch diverged — which shouldn't affect a local _commit_ operation, but depending on how the `git_commit` action is implemented in Fastlane core, maybe it does…?)